engine/runtime/src/main/java/org/enso/interpreter/node/callable/InvokeMethodNode.java

```diff
@@ -134,13 +138,59 @@ Object doPanicSentinel(
     throw self;
   }
 
-  @Specialization
+  @Specialization(guards = "warnings.hasWarnings(self)")
   Object doWarning(
       VirtualFrame frame,
       State state,
       UnresolvedSymbol symbol,
-      WithWarnings self,
-      Object[] arguments) {
+      Object self,
+      Object[] arguments,
+      @Cached MethodResolverNode methodResolverNode,
+      @CachedLibrary(limit = "10") WarningsLibrary warnings) {
+    Function function =
+        symbol
+            .getScope()
+            .lookupMethodDefinition(
+                EnsoContext.get(this).getBuiltins().warning().getEigentype(), symbol.getName());
+    if (function != null) {
+      // Warning's builtin type methods are static meaning that they have a synthetic `self`
```

Comment:
@JaroslavTulach Would very much appreciate your feedback on this bit. Unfortunately due to synthetic `self` inserted for static methods I have to somehow convince runtime that it is OK to call this function and it is fully applied. I figured that we are dealing with Warnings it's OK to hit this slow path.

Reply:
Where is the slow path? Only when creating the node? That's obviously have to be slow path. However it'd be better to have this AST manipulation be done automatically by the Truffle DSL.